rbc/irtools.py

```diff
@@ -162,7 +161,50 @@ def init(self):
 
     def load_additional_registries(self):
         # libdevice and math from cuda have precedence over the ones from CPU
-        if get_version('numba') >= (0, 49):
+        nb_version = get_version('numba')
+        if nb_version >= (0, 52):
+            from numba.cuda import libdeviceimpl, mathimpl
+            from .omnisci_backend import cuda_npyimpl
+            self.install_registry(libdeviceimpl.registry)
+            self.install_registry(mathimpl.registry)
+            self.install_registry(cuda_npyimpl.registry)
+        else:
+            import warnings
+            warnings.warn("libdevice bindings requires Numba 0.52 or newer,"
+                          f" got Numba v{'.'.join(map(str, nb_version))}")
+        super().load_additional_registries()
+
+
+class RemoteGPUTypingContext(typing.Context):
+    def load_additional_registries(self):
+        if get_version('numba') >= (0, 52):
+            from numba.core.typing import npydecl
+            from numba.cuda import cudamath, libdevicedecl
+            self.install_registry(npydecl.registry)
+            self.install_registry(cudamath.registry)
+            self.install_registry(libdevicedecl.registry)
+        super().load_additional_registries()
+
+# ---------------------------------------------------------------------------
+# Code generation methods
+
+
+class RemoteGPUTargetContext(cpu.CPUContext):
```

Comment:
The classes RemoteGPUTypingContext and RemoteGPUTargetContext are defined twice in this file.
Cleanup?